pyswarms/utils/search/grid_search.py

```diff
@@ -48,63 +48,12 @@
 import operator as op
 import itertools
 import numpy as np
+from pyswarms.utils.search.base_search import SearchBase
 
-class GridSearch(object):
+class GridSearch(SearchBase):
     """Exhaustive search of optimal performance on selected objective function
     over all combinations of specified hyperparameter values."""
```

Comment:
Hmmm... I may be wrong or my GitHub is not working properly, but does this need to have an `__init__` method similar to `RandomSearch()`? 😕 😄 

You also mentioned about the possibility that `GridSearch` will take so long when we're looking for a  lot of hyperparameters and take too many iterations 👍 . Maybe we can add a `logger.warn?` check the implementations in `GlobalBestPSO` and `LocalBestPSO`. 

There is a `logger.info('My information')` call in place of `print()` functions. I also initialized it with `logger = logging.getlogger(__name__)`.  You can do the same thing but call `logger.warn()` instead. 

Thus we can have something like:

```python
if condition_is_met:
    logger.warn('Using GridSearch will take so much time') 
# of course you can write a better warning than mine hehe
```

This will still run a search, only with an additional warning.

Reply:
Ha! Yes, I think I thought I was being clever because if no new attributes are added in the GridSearch `__init__` it can just just inherit SearchBase's `__init__`, which will be retrieved when creating a GridSearch instance. 

```
+  57     def __init__(self, optimizer, n_particles, dimensions, options,
+  58                  objective_func, iters,bounds=None, velocity_clamp=None):
+  59         """Initializes the paramsearch."""
+  60
+  61         # Assign attributes
+  62         super().__init__(optimizer, n_particles, dimensions, options,
+  63                 objective_func, iters, bounds=bounds,
+  64                 velocity_clamp=velocity_clamp)
```

Reply:
Thanks for pointing me to 'logger.warning'. In looking at `search` I realized I'm not sure how much of an issue run-time will be, even for GridSearch with lots of hyperparameter options.  

What I'd like to do next is write up an example for the docs on `Tuning PSO hyperparameters` that these two methods in action, and use that writing as a chance to read the literature a bit more on conventions for setting PSO hyperparameters. I think this will help me have a better sense of kinds of situations might come up. What I can see happening is including `logger.warn`, yet also having clear documentation that speaks to the conventions for setting `options` coming from the literature. My hunch was that runtime on big search spaces might need to be flagged for users, but I don't have a good sense of when a user would input a really long list for each of the 'c' weights, for instance.

Reply:
Actually, I realized that it's kinda hard to pinpoint when to issue a warning based on user input. I believe your plan is good for now 👍 

Regarding the `__init__`, the reason why I often add the `super()` is that it enables me to pass defaults into the base class. So just in case I want defaults in `GridSearch` etc., I just declare them on `GridSearch`'s init, then do the super.

For now I guess it is fine. I'd merge this implementation now 💯